src/ogdc_runner/recipe/simple.py

```diff
@@ -64,10 +89,132 @@ def _cmds_from_simple_recipe(recipe_dir: str) -> list[str]:
     return commands
 
 
-def make_simple_workflow(recipe_dir: str) -> Workflow:
-    """Run the workflow and return its name as a str."""
-    commands = _cmds_from_simple_recipe(recipe_dir)
-    recipe_config = get_recipe_config(recipe_dir)
+def _remove_existing_published_data(*, recipe_config: RecipeConfig) -> None:
+    """Executes an argo workflow that removes published data for a recipe if it
+    exists."""
+    with Workflow(
+        generate_name=f"{recipe_config.id}-remove-existing-data",
+        entrypoint="steps",
+        workflows_service=ARGO_WORKFLOW_SERVICE,
+    ) as w:
+        overwrite_template = Container(
+            name="overwrite-already-published",
+            command=["sh", "-c"],
+            args=[
+                f"rm -rf /mnt/{recipe_config.id}",
+            ],
+            volume_mounts=[
+                models.VolumeMount(
+                    name=OGDC_WORKFLOW_PVC.name,
+                    mount_path="/mnt/",
+                ),
+            ],
+        )
+        with Steps(name="steps"):
+            overwrite_template()
+
+    workflow_name = submit_workflow(workflow=w, wait=True)
+
+    # Cleanup this workflow, it is no longer needed
+    ARGO_WORKFLOW_SERVICE.delete_workflow(workflow_name)
+
+
+def _check_for_existing_published_data(*, recipe_config: RecipeConfig) -> bool:
```

Comment:
I'm beginning to wonder if we should standup a database that can keep track of recipes that have been submitted. The database could contain information on e.g., final data location, when the recipe was executed, recipe args/config, input data source, etc., which will help us with maintaining robust provenance for our derived datasets.

It would also simplify our approach to checking for published data. This function runs a relatively simple argo workflow to check for data on disk, but in the future we may also have other publication avenues we would want to check (e.g., a DataONE dataset). 

Instead of running an argo workflow to check if a recipe has been run, we would just perform a database query and check the status of a submitted recipe that way.

Reply:
For our MetaDIG quality engine, which has a similar architecture of jobs needing to run, we keep a separate database of results, which makes it fast and easy to find out what is in flight, what is complete, and to summarize the results/stats about runs. I think the question here is what use cases drive need for this info.